packages/pyright-internal/src/analyzer/sourceFile.ts

```diff
@@ -550,17 +548,36 @@ export class SourceFile {
             return undefined;
         }
 
-        assert(this._writableData.parserOutput !== undefined && this._writableData.parsedFileContents !== undefined);
+        assert(this._writableData.parserOutput !== undefined);
```

Comment:
we probably need to audit every places `getParseResults` is used. I believe we have many places where if `getParseResults` returned  `a value` once, we assume `getparseResults` will always return a value. now that assumption is not valid since it could return `undefined` next time `getParseResults` is called.

ex)
const parseResults = xxxx.getParseResults();
if (!parseResults) {
return ....
}

after this point,  I believe we have some code that will do `xxx.getParseresults()!` instead of passaround `parseResults` on every calls.

Reply:
we might add `getParserOutput` for consumers that only care about parse tree but not tokens. 

that said, for it to be useful, we probably need some data from `TokenizerOutput`? I think the memory heavy parts is `// List of all tokens. tokens: TextRangeCollection<Token>;` but other data are cheap to hold onto memory?

I wonder when `Diagnostic mode` is set to `workspace`, how UX will be like? it probably need to re-tokenize file for every diagnostics exist in closed files? not to get `token` but for `lines` and `pyright/type ignores` ?

it might be better to drop only `tokens` but keep `TokenizerOutput` ?